### PR TITLE
Hello world for pybind

### PR DESCRIPTION
**Pull request by icui**
# Description

Adds a python module when performing ```pip install```. The python package is located in python/specfempp, which follows the convention for pybind. In the future we can change this folder into a submodule.

# Issue Number

#182 

# Checklist

Please make sure to check developer documentation on specfem docs. 

- [x] I ran the code through pre-commit to check style
- [x] My code passes all the integration tests
- [x] I have added sufficient unittests to test my changes
- [x] I have added/updated documentation for the changes I am proposing
- [x] I have updated CMakeLists to ensure my code builds
- [x] My code builds across all platforms

### CODE REVIEW

**Review by Rohit-Kakodkar — COMMENTED** [review]
Why do we need to create a seperate driver file `specfem2d_py.cpp`? Can we not add the bindings to the same file?

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,13 +15,26 @@ set(NUM_CHUNKS 1)
...
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "IntelLLVM")
+            set(CMAKE_CXX_FLAGS "-fp-model=precise -fPIC")
+    elseif(CMAKE_CXX_COMPILER_ID STREQUAL "Intel")
+            message("-- Detected Intel classic compiler which will be deprecated soon.")
+            message("-- It is recommended you use IntelLLVM compiler.")
+            set(CMAKE_CXX_FLAGS "-diag-disable=10441 -fp-model=precise -fPIC")
+    else()
+            set(CMAKE_CXX_FLAGS "-fPIC")
+    endif()
+else (SPECFEMPP_ENABLE_PYTHON)
```
Comment (Rohit-Kakodkar) [inline CMakeLists.txt:29]: Is this syntax correct?

**Comment by icui** [comment]
> Why do we need to create a seperate driver file `specfem2d_py.cpp`? Can we not add the bindings to the same file?
Fixed now.

File: CMakeLists.txt (hunk 1 above)
Comment (icui) [reply CMakeLists.txt:29]: I think so. I have tested with both pip and cmake.

File: CMakeLists.txt [hunk 2]
```diff
@@ -534,7 +547,35 @@ else (DOXYGEN_FOUND)
...
-if (SPECFEMPP_ENABLE_PYTHON)
+if (SPECFEMPP_USE_SKBUILD)
     install(TARGETS specfem2d DESTINATION ${SKBUILD_SCRIPTS_DIR})
     install(FILES ${CMAKE_BINARY_DIR}/xmeshfem2D DESTINATION ${SKBUILD_SCRIPTS_DIR} PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE)
+endif (SPECFEMPP_USE_SKBUILD)
+
+if (SPECFEMPP_ENABLE_PYTHON)
+    find_package(Python REQUIRED COMPONENTS Interpreter Development.Module)
+    find_package(pybind11 CONFIG REQUIRED)
+    python_add_library(_core MODULE src/specfem2d.cpp WITH_SOABI)
```
Comment (Rohit-Kakodkar) [inline CMakeLists.txt:558]: Is this just taking the bindings from `src/specfem2d.cpp` and then compiling it under `_core`

Comment (icui) [reply CMakeLists.txt:558]: Yes

**Review by Rohit-Kakodkar — COMMENTED** [review]
> > Why do we need to create a seperate driver file `specfem2d_py.cpp`? Can we not add the bindings to the same file?

Fixed now. 

Eventually we'd want to move bindings to a `driver` module that runs specfem. And call the driver in main

**Review by Rohit-Kakodkar — COMMENTED** [review]
Otherwise this looks good

File: CMakeLists.txt (hunk 1 above)
Comment (lsawade) [reply CMakeLists.txt:29]: I actually like this syntax. It clears up what conditional you are in, and nothing would be put after else anyways.

I haven't seen it like that either though.

**Review by lsawade — APPROVED** [review]
Looks good to me

**Pull request merged**